From libc-ports-return-1325-listarch-libc-ports=sources dot redhat dot com at sourceware dot org Tue Aug 25 12:36:31 2009
Return-Path: <libc-ports-return-1325-listarch-libc-ports=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-libc-ports at sources dot redhat dot com
Received: (qmail 25364 invoked by alias); 25 Aug 2009 12:36:31 -0000
Received: (qmail 25356 invoked by uid 22791); 25 Aug 2009 12:36:30 -0000
X-SWARE-Spam-Status: No, hits=-1.9 required=5.0 	tests=AWL,BAYES_00,SPF_PASS
X-Spam-Check-By: sourceware.org
Received: from mail.codesourcery.com (HELO mail.codesourcery.com) (65.74.133.4)     by sourceware.org (qpsmtpd/0.43rc1) with ESMTP; Tue, 25 Aug 2009 12:36:24 +0000
Received: (qmail 15451 invoked from network); 25 Aug 2009 12:36:22 -0000
Received: from unknown (HELO mbp.local) (maxim@127.0.0.2)   by mail.codesourcery.com with ESMTPA; 25 Aug 2009 12:36:22 -0000
Message-ID: <4A93DABC.7080907@codesourcery.com>
Date: Tue, 25 Aug 2009 16:36:12 +0400
From: Maxim Kuvyrkov <maxim at codesourcery dot com>
User-Agent: Thunderbird 2.0.0.23 (Macintosh/20090812)
MIME-Version: 1.0
To: libc-ports at sourceware dot org
CC: Andreas Schwab <schwab at linux-m68k dot org>
Subject: [M68K/ColdFire patch 1/n] Update sysdep.h
References: <4A93D885.3060004@codesourcery.com>
In-Reply-To: <4A93D885 dot 3060004 at codesourcery dot com>
Content-Type: multipart/mixed;  boundary="------------060104060208030403060300"
X-IsSubscribed: yes
Mailing-List: contact libc-ports-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Id: <libc-ports.sourceware.org>
List-Subscribe: <mailto:libc-ports-subscribe at sourceware dot org>
List-Post: <mailto:libc-ports at sourceware dot org>
List-Help: <mailto:libc-ports-help at sourceware dot org>, <http://sourceware dot org/lists dot html#faqs>
Sender: libc-ports-owner at sourceware dot org
Delivered-To: mailing list libc-ports at sourceware dot org

This is a multi-part message in MIME format.
--------------060104060208030403060300
Content-Type: text/plain; charset=ISO-8859-1; format=flowed
Content-Transfer-Encoding: 7bit

This simple patch updates sysdep.h.

The definition of NEED_STATIC_SYSINFO_DSO at the bottom of the file is a 
no-op at the moment, but it serves as a helpful remainder for future 
improvement.  The improvement would be to implement linking in kernel 
vDSO for static executables.

Any comments?  OK to commit?

Thanks,

--
Maxim K.
CodeSourcery

--------------060104060208030403060300
Content-Type: text/plain; x-mac-type="0"; x-mac-creator="0";
 name="0001-Update-sysdep.h.patch"
Content-Transfer-Encoding: 7bit
Content-Disposition: inline;
 filename="0001-Update-sysdep.h.patch"

>From d5d20e06bf49c6586ce2e64e627c25d825341324 Mon Sep 17 00:00:00 2001
From: Maxim Kuvyrkov <maxim@codesourcery.com>
Date: Mon, 24 Aug 2009 12:28:47 +0400
Subject: [PATCH 1/5] Update sysdep.h.

	* sysdeps/unix/sysv/linux/m68k/sysdep.h (tls.h): Include.
	(INTERNAL_SYSCALL): Convert to INTERNAL_SYSCALL_NCS.
	(PTR_MANGLE, PTR_DEMANGLE): Define.
	(NEED_STATIC_SYSINFO_DSO): Define.

Note: support for NEED_STATIC_SYSINFO_DSO is not yet implemented in the main
GLIBC tree.
---
 sysdeps/unix/sysv/linux/m68k/sysdep.h |   23 ++++++++++++++++++++---
 1 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/m68k/sysdep.h b/sysdeps/unix/sysv/linux/m68k/sysdep.h
index 12687d8..afc0e13 100644
--- a/ports/sysdeps/unix/sysv/linux/m68k/sysdep.h
+++ b/ports/sysdeps/unix/sysv/linux/m68k/sysdep.h
@@ -1,4 +1,5 @@
-/* Copyright (C) 1996, 1997, 1998, 2000, 2003, 2004, 2006 Free Software Foundation, Inc.
+/* Copyright (C) 1996, 1997, 1998, 2000, 2003, 2004, 2006, 2009
+   Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Written by Andreas Schwab, <schwab@issan.informatik.uni-dortmund.de>,
    December 1995.
@@ -23,6 +24,7 @@
 
 #include <sysdeps/unix/sysdep.h>
 #include <sysdeps/m68k/sysdep.h>
+#include <tls.h>
 
 /* Defines RTLD_PRIVATE_ERRNO.  */
 #include <dl-sysdep.h>
@@ -148,9 +150,11 @@ SYSCALL_ERROR_LABEL:							      \
 	arg 3		%d3	     call-saved
 	arg 4		%d4	     call-saved
 	arg 5		%d5	     call-saved
+	arg 6		%a0	     call-clobbered
 
    The stack layout upon entering the function is:
 
+	24(%sp)		Arg# 6
 	20(%sp)		Arg# 5
 	16(%sp)		Arg# 4
 	12(%sp)		Arg# 3
@@ -229,7 +233,7 @@ SYSCALL_ERROR_LABEL:							      \
    normally.  It will never touch errno.  This returns just what the kernel
    gave back.  */
 #undef INTERNAL_SYSCALL
-#define INTERNAL_SYSCALL(name, err, nr, args...)	\
+#define INTERNAL_SYSCALL_NCS(name, err, nr, args...)	\
   ({ unsigned int _sys_result;				\
      {							\
        /* Load argument values in temporary variables
@@ -237,7 +241,7 @@ SYSCALL_ERROR_LABEL:							      \
 	  before the call used registers are set.  */	\
        LOAD_ARGS_##nr (args)				\
        LOAD_REGS_##nr					\
-       register int _d0 asm ("%d0") = __NR_##name;	\
+       register int _d0 asm ("%d0") = name;		\
        asm volatile ("trap #0"				\
 		     : "=d" (_d0)			\
 		     : "0" (_d0) ASM_ARGS_##nr		\
@@ -245,6 +249,8 @@ SYSCALL_ERROR_LABEL:							      \
        _sys_result = _d0;				\
      }							\
      (int) _sys_result; })
+#define INTERNAL_SYSCALL(name, err, nr, args...)	\
+  INTERNAL_SYSCALL_NCS (__NR_##name, err, nr, ##args)
 
 #undef INTERNAL_SYSCALL_ERROR_P
 #define INTERNAL_SYSCALL_ERROR_P(val, err)		\
@@ -300,4 +306,15 @@ SYSCALL_ERROR_LABEL:							      \
 #define ASM_ARGS_6	ASM_ARGS_5, "a" (_a0)
 
 #endif /* not __ASSEMBLER__ */
+
+/* Pointer mangling is not yet supported for M68K.  */
+#define PTR_MANGLE(var) (void) (var)
+#define PTR_DEMANGLE(var) (void) (var)
+
+#if defined NEED_DL_SYSINFO || defined NEED_DL_SYSINFO_DSO
+/* M68K needs system-supplied DSO to access TLS helpers
+   even when statically linked.  */
+# define NEED_STATIC_SYSINFO_DSO 1
+#endif
+
 #endif
-- 
1.6.4


--------------060104060208030403060300--

